<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type" content="text/html;charset=utf-8">
  <title>QIIME</title>
<script src="https://www.google.com/jsapi?key=ABQIAAAAdyJBQQABvxRaME0S-wyQoxTvN-l145hOL_mSBP7Ok7BhV-E7XBQBTiu1ANdFaBO1ezmiKNOZBPvvzg" type="text/javascript"></script>
<script type="text/javascript" src="qiime.js"></script>
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ?
"https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost +
"google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-6636235-4");
pageTracker._trackPageview();
} catch(err) {}</script>
<link rel="stylesheet" type="text/css" href="qiime.css">
<link rel="shortcut icon" href="favicon.ico"> 
<body onload="loadHeaderFooter('dataFiles')">
</head>

<body>
<div class="wrapper clearfix">
	<div class="header" id="header"></div>
	<div class="leftcol" id="leftcol"></div>
	<div class="centercol">
	<h1> Data Files and Other Resources<img src="images/spacer_dots.png" alt="..."></h1>
	<h3> Data Files </h3>
	<ul>
		<li>
			<b><a href="http://greengenes.secondgenome.com/">Greengenes</a> OTUs (16S)</b><p>
			<a href="ftp://greengenes.microbio.me/greengenes_release/gg_13_5/gg_13_8_otus.tar.gz">13_8 (most recent)</a><p>
			<a href="ftp://greengenes.microbio.me/greengenes_release/gg_13_5/gg_13_5_otus.tar.gz">13_5</a><p>
			<a href="ftp://greengenes.microbio.me/greengenes_release/gg_12_10/gg_12_10_otus.tar.gz">12_10</a><p>
			<a href="http://greengenes.lbl.gov/Download/Sequence_Data/Fasta_data_files/Caporaso_Reference_OTUs/gg_otus_4feb2011.tgz">February 4th, 2011</a>
		</li>
		<hr>
		<li>
			<b><a href="http://www.arb-silva.de/">Silva</a> OTUs (16S/18S)</b><p>
			<a href="ftp://thebeast.colorado.edu/pub/QIIME_nonstandard_referencedb/Silva_111.tgz">Silva release 111 (most recent)</a><p>
			<a href="http://www.arb-silva.de/download/archive/qiime/">Silva release 108</a><p>
			<a href="ftp://thebeast.colorado.edu/pub/QIIME_nonstandard_referencedb/Silva_108_database_curated.tgz">Silva 108 database curated to reflect eukaryotic phylogeny</a> and <a href="ftp://thebeast.colorado.edu/pub/QIIME_nonstandard_referencedb/Silva_108_Qiime_format_notes.txt">associated notes</a>.
		</li>
		<hr>
		<li>
			<b><a href="http://unite.ut.ee/">UNITE</a> OTUs (ITS)</b><p>
			<a href="http://unite.ut.ee/repository.php">All releases, including the latest, are available for download from the UNITE website here</a><p>
			<a href="https://github.com/downloads/qiime/its-reference-otus/its_12_11_otus.tar.gz">12_11</a> (alpha release*)<br>
			<table><tr><td bgcolor=FF6666><font size=-2>
			*These reference sequence sets represent de-replicated (clustered) versions (at 99% and 97% sequence similarity) of all fungal rDNA ITS sequences in the current <a href="http://unite.ut.ee/">UNITE</a>+INSD (International Nucleotide Sequence Databases: NCBI, EMBL, DDBJ) release of circa 300,000 sequences. The taxonomy mapping files provided here were created from the Index Fungorum ranked classification schema provided by <a href="http://unite.ut.ee/">UNITE</a> in association with each sequence in their database.  Efforts have been made by <a href="http://unite.ut.ee/">UNITE</a> to improve the taxonomic information associated with some of the sequences in their database.  The QIIME reference sequence sets linked here have not been subject to any other form of curation (manual or automated) and certainly include incorrectly identified sequences, chimeras, and other problematic sequences.  The data provide here are a fungal ITS reference alpha version. Responsibility lies with the user to verify the accuracy of associated taxonomic information as well as quality of sequences in these datasets.  Improved fungal rDNA ITS reference sets based on the semi-curated centroids for sequence clusters in the <a href="http://unite.ut.ee/">UNITE</a> Global Key Annotations module will soon be available here.<br></font></tr></td></table><p>
			<a href="http://tajmahal.colorado.edu/tmp/unite_taxonomy_21nov2011.zip">21Nov2011</a>
		</li>
		<hr>
		<li>
			<b><a href="http://img.jgi.doe.gov/cgi-bin/w/main.cgi">IMG</a>/QIIME reference protein sequences</b><p>
			<a href="ftp://thebeast.colorado.edu/img-qiime-25oct2012.tgz">img-qiime-25oct2012</a> (alpha release; most recent)
		</li>
		</ul>
		<h3>Latest Amazon Web Services (AWS) virtual machine identifiers</h3>
		<ul>
		<li>QIIME 1.8.0 (release) AMI: ami-e788a08e</li>
		</ul>
		<h3>Older Amazon Web Services (AWS) virtual machine identifiers</h3>
		<ul>
		<li>QIIME 1.7.0 (release) AMI: ami-9bc9a7f2</li>
		<li>QIIME 1.6.0 (release) AMI: ami-69c34c00</li>
		<li>QIIME 1.5.0 (release) AMI: ami-e4bf1b8d</li>
		<li>QIIME 1.4.0 (release) AMI: ami-458d5b2c</li>
		<li>QIIME 1.4.0 (release) Extra-large AMI: ami-438d5b2a</li>
		<li>QIIME 1.4.0-dev (development) Extra-large AMI: ami-3461bd5d</li>
		<li>QIIME 1.3.0 (release) AMI: ami-dce51eb5</li>
		<li>QIIME 1.3.0-dev (development) AMI: ami-9ba56bf2</li>
		</ul>
		<h3>To upgrade the release AMI to the latest development version of QIIME, please use the <a href="https://github.com/qiime/qiime-deploy">qiime-deploy</a> tool.</h3>
		<h3> Protocols </h3>
		<ul>
			<li>Primers and amplification protocols for 454 FLX+ long read 16S and 18S amplicons. (version <a href="https://s3.amazonaws.com/s3-qiime_tutorial_files/454_FLXplus_protocols_13_4.tgz">13_4</a>)
			<li>16S Illumina HiSeq/MiSeq Protocols: <a href="http://www.nature.com/ismej/journal/v6/n8/full/ismej20128a.html">Caporaso et al, 2011</a> and <a href="http://www.earthmicrobiome.org/emp-standard-protocols/16s/">EMP protocols page</a>
			<li>16S Illumina GAIIx Protocols (<b>these are different from those for the HiSeq/MiSeq!</b>): <a href="http://www.pnas.org/content/early/2010/06/02/1000080107">Caporaso et al, 2010</a>
		</ul>
</div>
<div class="footer" id="footer"></div>
</div>
</body>
</html>
