<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type" content="text/html;charset=utf-8">
  <title>QIIME</title>
<script src="https://www.google.com/jsapi?key=ABQIAAAAdyJBQQABvxRaME0S-wyQoxTvN-l145hOL_mSBP7Ok7BhV-E7XBQBTiu1ANdFaBO1ezmiKNOZBPvvzg" type="text/javascript"></script>
<script type="text/javascript" src="home_static/qiime.js"></script>
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ?
"https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost +
"google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-6636235-4");
pageTracker._trackPageview();
} catch(err) {}</script>
<link rel="stylesheet" type="text/css" href="home_static/qiime.css">
<link rel="shortcut icon" href="favicon.ico"> 
<body onload="loadHeaderFooterRoot('main')">
</head>

<body>
	<a href="https://github.com/qiime/qiime"><img style="position: absolute; top: 0; left: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_left_green_007200.png" alt="Fork me on GitHub"></a>
	<div class="wrapper">
		<div class="header" id="header">
		</div>
		<div class="leftcol" id="leftcol">
		</div>
		<div class="centercol">
			<!-- <div class="content" id="content"> -->
				<div>
					<h1> What is QIIME? <img src="home_static/images/spacer_dots.png" alt="..."></h1>
					<p>
						QIIME (canonically pronounced "chime") stands for Quantitative Insights Into
		Microbial Ecology. QIIME is an open source software package for
		comparison and analysis of microbial communities, primarily based on
		high-throughput amplicon sequencing data (such as SSU rRNA) generated
		on a variety of platforms, but also supporting analysis of other types
		of data (such as shotgun metagenomic data). QIIME takes users from
		their raw sequencing output through initial analyses such as OTU
		picking, taxonomic assignment, and construction of phylogenetic trees
		from representative sequences of OTUs, and through downstream
		statistical analysis, visualization, and production of
		publication-quality graphics. QIIME has been applied to studies
		based on billions of sequences from thousands of samples.
					</p>
				</div>
				<div>
					<h2> The ISME 15 QIIME Workshop</h2>
					<p>The QIIME Developers will be teaching a workshop after <a href="http://www.isme-microbes.org/isme15">ISME 15</a> on August 30th, 2014 in Seoul, South Korea. This is the Saturday following ISME, and will be held at Seoul National University, near the ISME venue.</p>
					<p><a href="http://qiime.org/workshops/isme15-qiime.pdf">Details on the workshop, including a link to the application.</a> <b>Application deadline <s>is April 22, 2014</s> has been extended to May 20, 2014.</b></p>
				</div>
				<div>
					<h2> Getting started with QIIME </h2>
					<p><b>Installing:</b> The quickest way to get started using QIIME is with <a href="http://www.wernerlab.org/software/macqiime">MacQIIME</a> (if you're running Mac OS X), <a href="https://github.com/qiime/qiime-deploy">qiime-delpoy</a> (if you're using Linux), or the <a href="http://qiime.org/install/vm_ec2.html">Amazon EC2 image</a> or <a href="http://qiime.org/install/virtual_box.html">VirtualBox</a> (if you're using Windows, Mac OS X, or Linux).</p>
					<p><b>Running:</b> Once you've installed QIIME, move on to the <a href="http://qiime.org/tutorials/index.html">QIIME Tutorials</a>. The <a href="http://qiime.org/tutorials/illumina_overview_tutorial.html">Illumina overview tutorial</a> is a good first analysis to run. In this tutorial you'll download a small data set and work through a series of commands that will introduce you to some of QIIME's commonly used features and analyses.</p>
					<p><b>Getting help:</b> We provide support for QIIME on the <a href="http://forum.qiime.org">QIIME Forum</a>. Before requesting help with QIIME on the forum, please review <a href="https://groups.google.com/forum/#!topic/qiime-forum/Z-KrSvdWVa4">this post</a>. We devote a lot of time to helping users on the forum, and reviewing that post will help us to efficiently help you, leaving us with more time for QIIME development and our own research.</p>
					<p>For getting started on interacting with the command line, please
review <a href="https://groups.google.com/forum/#!topic/qiime-forum/gs7v73s61Pg">this post</a>.</p>
					<p><b>QIIME scripts:</b> As you become more proficient with QIIME, <a href="http://qiime.org/scripts/index.html">the QIIME script documentation</a> will help you explore and learn QIIME's functionality.</p>
					<h2> Code </h2>
					<p>
					QIIME is open source software. You can find the code under public revision control in our <a href="https://github.com/qiime/qiime">GitHub repository</a>. You can find related projects under the <a href="https://github.com/qiime">QIIME organization on GitHub</a>. We're very interested in community contributions and feedback. Use <a href="https://help.github.com/articles/using-pull-requests">pull requests</a> to contribute code or documentation, and our <a href="https://github.com/qiime/qiime/issues">issue tracker</a> to report bugs or request new features.
					</p>
					<h3> Other ways to use QIIME </h3>
					<p>
						<a href="https://github.com/qiime/qiime-deploy">QIIME-deploy</a>: Install QIIME easily on Ubuntu and RedHat Linux.
<br />
						<a href="http://www.wernerlab.org/software/macqiime">MacQIIME</a>: Easy install of QIIME on MacOS X.
<br />
						<a href="http://nebc.nerc.ac.uk/tools/bio-linux">Bio-Linux</a>: QIIME and hundreds of other bioinformatics tools in an Ubuntu-based Linux distribution.
<br />
						<a href="http://www.n3phele.com">n3phele</a>: Run QIIME on the Amazon Cloud from a web interface - no command line interaction is required.
					</p>
				</div>
				<div>
					<h2> News </h2>
					<div id="news">
					</div>
				</div>
				<div>
					<h2> Latest forum posts </h2>
					<div id="forum">
					</div>
				</div>
				<div>
					<h2> Most recent articles citing QIIME </h2>
					<div id="citations">
					</div>
				</div>
				<div>
					<h2>Citing QIIME<a class="headerlink" href="#citing-qiime" title="Permalink to this headline"></a></h2>
					<p>If you use QIIME for any published research, please include the following citation:</p>
					<blockquote>
						<div>
							<p><strong>QIIME allows analysis of high-throughput community sequencing data</strong>
							</p>
							<p>J Gregory Caporaso, Justin Kuczynski, Jesse Stombaugh, Kyle Bittinger, Frederic D Bushman, Elizabeth K Costello, Noah Fierer, Antonio Gonzalez Pena, Julia K Goodrich, Jeffrey I Gordon, Gavin A Huttley, Scott T Kelley, Dan Knights, Jeremy E Koenig, Ruth E Ley, Catherine A Lozupone, Daniel McDonald, Brian D Muegge, Meg Pirrung, Jens Reeder, Joel R Sevinsky, Peter J Turnbaugh, William A Walters, Jeremy Widmann, Tanya Yatsunenko, Jesse Zaneveld and Rob Knight; Nature Methods, 2010; doi:10.1038/nmeth.f.303
							</p>
						</div>
					</blockquote>
					<p>QIIME wraps many other software packages, and these should be cited if they are used. For example, users of <tt>pick_de_novo.py</tt> are actually using QIIME, uclust, RDP classifier, PyNAST, and FastTree, since QIIME is wrapping those applications. <b>Any time you're using tools that QIIME wraps, it is essential to cite those tools.</b> The Werner Lab website <a href="http://www.wernerlab.org/software/macqiime/citations">has a list of software, including citations, that is used in the various stages of QIIME</a>.
					<p>You can find the <a class="reference external" href="http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.f.303.html">QIIME paper here</a>, and the data presented in this paper can be found 
						<a class="reference external" href="http://bmf.colorado.edu/QIIME/QIIME_NM_2010.tgz">here</a>.
					<br><br>
					</p>

				</div>

			<!-- </div> -->
		</div>
		<div class="footer clearfix" id="footer">
		</div>
	</div>
</body>
</html>
