<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Processing Illumina Data &mdash; Homepage</title>
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0.0-dev',
        COLLAPSE_MODINDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
    <link rel="top" title="Homepage" href="../index.html" />     
<meta http-equiv="Content-Style-Type" content="text/css" />
<script type="text/javascript" src="http://www.google.com/jsapi?key=ABQIAAAAbW_pA971hrPgosv-Msv7hRRE2viNBUPuU405tK6p2cguOFmlFBQSwZMG6_q_v6Z42nkdo9ejT1aHmA"></script>
<script type="text/javascript" src="../_static/google_feed.js"></script>

  </head>
  <body>

<div style="background-color: white; text-align: left; padding: 0px">
<a href="../index.html"><img src="../_static/wordpressheader.png" border="0" alt="sampledoc"/></a>
</div>
<div class="news" style="background-color:#ccc;"><table style="font-size:12pt;padding: 5px;"><tr id="feed"><td><b>News and Announcements &raquo;</b></td>
</tr></table></div>

    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="../index.html">Home</a> &raquo;</li> 
      </ul>
    </div>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="processing-illumina-data">
<span id="id1"></span><h1>Processing Illumina Data<a class="headerlink" href="#processing-illumina-data" title="Permalink to this headline">¶</a></h1>
<p>This document describes how to process Illumina sequencing data with QIIME. As the Illumina platform is just beginning to be used for community sequencing, the Illumina support in QIIME is still in development status. If you run into issues because your Illumina output is different from what we assume here, we recommend getting in touch via <a class="reference external" href="mailto:qiime&#46;help&#37;&#52;&#48;colorado&#46;edu">qiime<span>&#46;</span>help<span>&#64;</span>colorado<span>&#46;</span>edu</a>. Full support for analyzing community sequencing data generated on the Illumina platform is one of our goals, so we&#8217;re willing to help out.</p>
<p>QIIME can be used to parse single-end or paired-end read data from the Illumina platform. The downstream support for analysis of paired-end read data is currently more limited. The parsed output is in standard fasta format, so all scripts (such as align_seqs.py and assign_taxonomy.py) can read it. However because there may be a &#8216;big gap&#8217; between the 5&#8217; and 3&#8217; reads if the primers are distant in the sequence, or conversely because the reads may overlap if the primers are close, your mileage with the downstream tools may vary.</p>
<p>This example illustrates how to used <a class="reference external" href="../scripts/split_libraries_illumina.html">split_libraries_illumina.py</a> to parse your Illumina output into a format that can be used by QIIME.</p>
<div class="section" id="input-file-formats">
<h2>Input file formats<a class="headerlink" href="#input-file-formats" title="Permalink to this headline">¶</a></h2>
<p>The input to split_libraries_illumina.py is one or more <tt class="docutils literal"><span class="pre">sequence</span></tt> files, and a standard QIIME mapping file. For example, 10 lines extracted from a <tt class="docutils literal"><span class="pre">s_1_1_sequence.txt</span></tt> file (where the first <tt class="docutils literal"><span class="pre">1</span></tt> refers to the lane, and the second <tt class="docutils literal"><span class="pre">1</span></tt> refers to the read, indicating the 5&#8217; read in this case) are:</p>
<div class="highlight-python"><pre>HWI-6X_9267:1:1:12:410#ACAGCTA/1:TACGTAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGCATTTTAAGCCAGACGTGAAATCCCCGGGCTTAACCTGGGAACTG:abbb`aaa`^aa```ba`aaaabaaaabaaaa^[Y]^__a`abb`aaaa]Y\\_a[Y_a`a```a__]aaXT\`^\_]`a^^WSZ\JNY]^a`ORO^^`Y
HWI-6X_9267:1:1:12:1762#ACATGAT/1:GACGGAGGATGCAAGTGTTATCCGGAATCACTGGGCGTAAAGCGTCTGTAGGTTGTTTGATAAGTCAACTGTTAAATCTTGAAGCTCAACTTCAAAATCG:aaaaaaaaabaaaaa_aaaaaa`aaaaaaaa`aa``a]aa```a^a^`\```\a`^aaa_\__]]_a_``^``a^^a^b[`SJN]Y_ZZ]^W___`_^U[
HWI-6X_9267:1:1:12:1872#ACAGTTG/1:TACGGAGGGGGTTAGCGTTGTTCCGAATTACTGGGCGTAAAGCGCGCGTAGGCGGATTAGAAAGTTGGGGGGGAAATCCCGGGGCTCAACCCCGGACGTG:aaaaa_aaaa`[a_a`aaaa]a[MY``a\a`aaaaa_\]_\__[_]W]^[[U]aXRZ\W[J\KVTX]\YZZDVY]SUBBBBBBBBBBBBBBBBBBBBBBB</pre>
</div>
<p>You&#8217;ll notice that each line contains seven <tt class="docutils literal"><span class="pre">:</span></tt> delimited fields. These are:</p>
<blockquote>
<ol class="arabic simple">
<li>Machine name</li>
<li>Channel/lane number</li>
<li>Tile number</li>
<li>X position</li>
<li>Y position</li>
<li>nucleotide sequence</li>
<li>quality score</li>
</ol>
</blockquote>
<p>In this case, the <tt class="docutils literal"><span class="pre">Y</span> <span class="pre">position</span></tt> field additionally contains the reverse compliment of the barcode sequence as the first <tt class="docutils literal"><span class="pre">B</span></tt> bases following the <tt class="docutils literal"><span class="pre">#</span></tt> sign, where <tt class="docutils literal"><span class="pre">B</span></tt> refers to the barcode length as determined from the mapping file. If your data looks different from this, get in touch with <a class="reference external" href="mailto:qiime&#46;help&#37;&#52;&#48;colorado&#46;edu">qiime<span>&#46;</span>help<span>&#64;</span>colorado<span>&#46;</span>edu</a>, and we can try to work on custom solutions until standards are developed in this area.</p>
<p>One important thing to note is that our sequencing primers were developed to avoid sequencing the PCR primers. So unlike standard 454 data, there are no barcode, linker, or primer sequences that need to be extracted from these sequences. As a consequence, the <tt class="docutils literal"><span class="pre">LinkerPrimerSequence</span></tt> in the mapping file is not important. In the following example, we&#8217;ve simply included an identifier for the LinkerPrimerSequence. An example mapping file to match this data might look like:</p>
<div class="highlight-python"><pre>#SampleID       BarcodeSequence LinkerPrimerSequence    SampleType      Description
Sample1 AGCTGT  ILBC_17 Freshwater      creek_3-4_cm_depth
S2      AACTGT  ILBC_19 Ocean   ocean_2-3_cm_depth
S3      TCATGT  ILBC_21 Ocean   ocean_3-4_cm_depth
sample4 GCTGGT  ILBC_26 Feces   fecal_subject_5</pre>
</div>
</div>
<div class="section" id="parsing-5-read-data">
<h2>Parsing 5&#8217; read data<a class="headerlink" href="#parsing-5-read-data" title="Permalink to this headline">¶</a></h2>
<p>To process a single lane of 5&#8217; read Illumina data, you would run the command:</p>
<div class="highlight-python"><pre>split_libraries_illumina.py -5 s_1_1_sequence.txt -o 5prime/ -m illumina_mapping.txt</pre>
</div>
<p>Similarly, to process multiple lanes of 5&#8217; reads of Illumina data, you would run the command:</p>
<div class="highlight-python"><pre>split_libraries_illumina.py -5 s_1_1_sequence.txt,s_2_1_sequence.txt -o 5prime/ -m illumina_mapping.txt</pre>
</div>
<p>Note that the sequence filepaths are comma-separated. There can be no spaces between the filepaths and the comma(s). This would result in four output files in <tt class="docutils literal"><span class="pre">5prime/</span></tt>:</p>
<blockquote>
<ol class="arabic simple">
<li><tt class="docutils literal"><span class="pre">s_1_5prime_seqs.fasta</span></tt>: lane 1 sequences which have passed quality filtering</li>
<li><tt class="docutils literal"><span class="pre">s_1_5prime_qual.txt</span></tt>: lane 1 quality scores for sequences which have passed quality filtering</li>
<li><tt class="docutils literal"><span class="pre">s_2_5prime_seqs.fasta</span></tt>: lane 2 sequences which have passed quality filtering</li>
<li><tt class="docutils literal"><span class="pre">s_2_5prime_qual.txt</span></tt>: lane 2 quality scores for sequences which have passed quality filtering</li>
</ol>
</blockquote>
<p>If you wish to combine the data from different lanes for downstream analysis, you can do this using <tt class="docutils literal"><span class="pre">cat</span></tt>. For example:</p>
<div class="highlight-python"><pre>cat s_1_5prime_seqs.fasta &gt;&gt; s_5prime_seqs.fasta
cat s_2_5prime_seqs.fasta &gt;&gt; s_5prime_seqs.fasta</pre>
</div>
</div>
<div class="section" id="parsing-3-read-data">
<h2>Parsing 3&#8217; read data<a class="headerlink" href="#parsing-3-read-data" title="Permalink to this headline">¶</a></h2>
<p>Parsing 3&#8217; read data is handled in the same way as parsing 5&#8217; read data, except that the <tt class="docutils literal"><span class="pre">-3</span></tt> parameter is passed in place of the <tt class="docutils literal"><span class="pre">-5</span></tt> parameter. For example, to process multiple lanes of 3&#8217; Illumina read data, you would run the command:</p>
<div class="highlight-python"><pre>split_libraries_illumina.py -3 s_1_2_sequence.txt,s_2_2_sequence.txt -o 3prime/ -m illumina_mapping.txt</pre>
</div>
<p>It is very important that you get the orientation correct, as the sequences are adjusted differently according to whether they are passed as <tt class="docutils literal"><span class="pre">-5</span></tt> or <tt class="docutils literal"><span class="pre">-3</span></tt>.</p>
</div>
<div class="section" id="parsing-paired-end-read-data">
<h2>Parsing paired-end read data<a class="headerlink" href="#parsing-paired-end-read-data" title="Permalink to this headline">¶</a></h2>
<p>To process a single lane of paired-end read Illumina data, you would run the command:</p>
<div class="highlight-python"><pre>split_libraries_illumina.py -5 s_1_1_sequence.txt -3 s_1_2_sequence.txt -o paired_end/ -m illumina_mapping.txt</pre>
</div>
<p>To process multiple lanes of paired-end read Illumina data, you would run the command:</p>
<div class="highlight-python"><pre>split_libraries_illumina.py -5 s_1_1_sequence.txt,s_2_1_sequence.txt -3 s_1_2_sequence.txt,s_2_2_sequence.txt -o paired_end/ -m illumina_mapping.txt</pre>
</div>
<p>Note that you must correctly match the order of the sequence file paths passed via <tt class="docutils literal"><span class="pre">-5</span></tt> and <tt class="docutils literal"><span class="pre">-3</span></tt>. In this case, the output sequence files will contain the 5&#8217; ends of the reads concatenated with the 3&#8217; ends of the reads. The orientations of the reads are adjusted so the full read makes sense. It is important to note however that aside from adjusting the read orientation, nothing is done to modify the read. So, for example, if your reads overlap there will be a repeat at the junction of the reads corresponding to the overlap.</p>
</div>
<div class="section" id="quality-filtering-of-illumina-data">
<h2>Quality filtering of Illumina data<a class="headerlink" href="#quality-filtering-of-illumina-data" title="Permalink to this headline">¶</a></h2>
<p>The methods currently included in QIIME for quality filtering Illumina data are presented in a publication that has been provisionally accepted. A link to that article will be provided here upon full acceptance. Briefly, a sequence is discarded if any of the following conditions are met:</p>
<blockquote>
<ul class="simple">
<li>The barcode contains one or more <tt class="docutils literal"><span class="pre">N</span></tt> bases, corresponding to ambiguous base calls.</li>
<li>The barcode is not an exact match to a barcode in the mapping file (to disable this, pass <tt class="docutils literal"><span class="pre">-u</span></tt>, which will cause the resulting sequences to be store with sample ID <tt class="docutils literal"><span class="pre">Unassigned</span></tt>.)</li>
<li>The sequence contains one or more <tt class="docutils literal"><span class="pre">N</span></tt> bases, corresponding to ambiguous base calls.</li>
<li>The high-quality region of the sequence is less than 75 bases long (adjustable with the <tt class="docutils literal"><span class="pre">-p</span></tt> parameter), where high-quality regions is defined a stretch of bases containing no more than 1 (adjustable with the <tt class="docutils literal"><span class="pre">-r</span></tt> parameter) quality score less than 1e-5 (adjustable with the <tt class="docutils literal"><span class="pre">-q</span></tt> parameter). In other words, with the default parameter settings, the read is truncated at the base preceding the first low quality stretch, and the truncated sequence must be greater than or equal to 75 bases long to be retained.</li>
</ul>
</blockquote>
</div>
<div class="section" id="using-illumina-data-in-downstream-analysis">
<h2>Using Illumina data in downstream analysis<a class="headerlink" href="#using-illumina-data-in-downstream-analysis" title="Permalink to this headline">¶</a></h2>
<p>Once you have processed your Illumina data with <a class="reference external" href="../scripts/split_libraries_illumina.html">split_libraries_illumina.py</a>, you can use the resulting files in all downstream QIIME scripts, including the workflow scripts. Currently the quality files (e.g., <tt class="docutils literal"><span class="pre">s_2_5prime_qual.txt</span></tt>) are not used anywhere in QIIME, but are provided to support more complex quality filtering. We are very interested in collaborating on developing better ways to quality filter Illumina data, so please feel free to get in touch if you have interest in this problem.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">

    
            <h3><a href="../index.html">Table Of Contents</a></h3>
            <ul>
<li><a class="reference external" href="">Processing Illumina Data</a><ul>
<li><a class="reference external" href="#input-file-formats">Input file formats</a></li>
<li><a class="reference external" href="#parsing-5-read-data">Parsing 5&#8217; read data</a></li>
<li><a class="reference external" href="#parsing-3-read-data">Parsing 3&#8217; read data</a></li>
<li><a class="reference external" href="#parsing-paired-end-read-data">Parsing paired-end read data</a></li>
<li><a class="reference external" href="#quality-filtering-of-illumina-data">Quality filtering of Illumina data</a></li>
<li><a class="reference external" href="#using-illumina-data-in-downstream-analysis">Using Illumina data in downstream analysis</a></li>
</ul>
</li>
</ul>


	<h3><a href="../index.html">Site index</a></h3>
	<ul><li><ul>
		
	 <li><a href="../index.html">Home</a><br /></li>
	 <li><a href="../install/index.html">Install</a><br /></li>
	 <li><a href="../documentation/index.html">Documentation</a><br /></li>
	 <li><a href="index.html">Tutorials</a><br /></li>
	 <li><a href="http://qiime.wordpress.com">Blog</a><br /></li>
	 <li><a href="../developer/index.html">Developer</a><br /></li>
	</ul></li></ul>


          <div id="searchbox" style="display: none">
            <h3>Quick search</h3>
              <form class="search" action="../search.html" method="get">
                <input type="text" name="q" size="18" />
                <input type="submit" value="Go" />
                <input type="hidden" name="check_keywords" value="yes" />
                <input type="hidden" name="area" value="default" />
              </form>
              <p class="searchtip" style="font-size: 90%">
              Enter search terms or a module, class or function name.
              </p>
          </div>
          <script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li><a href="../index.html">Home</a> &raquo;</li> 
      </ul>
    </div>

    <div class="footer">
      &copy; Copyright 2010, QIIME Team.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 0.6.3.
    </div>
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-6636235-4");
pageTracker._trackPageview();
} catch(err) {}</script>

  </body>
</html>