<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>make_phylogeny.py – Make Phylogeny &mdash; Homepage</title>
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.9',
        COLLAPSE_MODINDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
    <link rel="top" title="Homepage" href="../index.html" />
    <link rel="up" title="Scripts - Analyses and Parameters" href="../doc_script_index.html" />
    <link rel="next" title="assign_taxonomy.py – Assign taxonomy to each sequence" href="assign_taxonomy.html" />
    <link rel="prev" title="parallel_align_seqs_pynast.py – Parallel sequence alignment using PyNAST" href="parallel_align_seqs_pynast.html" /> 
  </head>
  <body>

<div style="background-color: white; text-align: left; padding: 0px">
<a href="../index.html"><img src="../_static/wordpressheader.png" border="0" alt="sampledoc"/></a>
</div>
<div class="news" style="background-color:#ccc;">
	<p style="font-size:12pt;padding: 5px;"><b>News and Announcements &raquo;</b>
<script type="text/javascript" src="http://app.feed.informer.com/digest3/U9OLMKHF9N.js"><noscript><a href="http://app.feed.informer.com/digest3/U9OLMKHF9N.html">Click for &quot;QIIME&quot;.</a> Powered by <a href="http://feed.informer.com/">RSS Feed Informer</a></noscript></script></p>
</div>

    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="assign_taxonomy.html" title="assign_taxonomy.py – Assign taxonomy to each sequence"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="parallel_align_seqs_pynast.html" title="parallel_align_seqs_pynast.py – Parallel sequence alignment using PyNAST"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Home</a> &raquo;</li>
          <li><a href="../documentation.html" >QIIME Documentation</a> &raquo;</li>
          <li><a href="../doc_script_index.html" accesskey="U">Scripts - Analyses and Parameters</a> &raquo;</li> 
      </ul>
    </div>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <span class="target" id="make-phylogeny"></span><div class="section" id="make-phylogeny-py-make-phylogeny">
<span id="index-111"></span><h1><em>make_phylogeny.py</em> &#8211; Make Phylogeny<a class="headerlink" href="#make-phylogeny-py-make-phylogeny" title="Permalink to this headline">¶</a></h1>
<p><strong>Description:</strong></p>
<p>Many downstream analyses require that the phylogenetic tree relating the OTUs in a study be present. The script <a class="reference external" href="./make_phylogeny.html">make_phylogeny.py</a> produces this tree from a multiple sequence alignment. Trees are constructed with a set of sequences representative of the OTUs, by default using FastTree (Price, Dehal, &amp; Arkin, 2009).</p>
<p><strong>Usage:</strong> <tt class="docutils literal"><span class="pre">make_phylogeny.py</span> <span class="pre">[options]</span></tt></p>
<p><strong>Input Arguments:</strong></p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p><strong>[REQUIRED]</strong></p>
<dl class="docutils">
<dt>-i, <cite>-</cite>-input_fp</dt>
<dd>Path to read input alignment</dd>
</dl>
<p><strong>[OPTIONAL]</strong></p>
<dl class="last docutils">
<dt>-t, <cite>-</cite>-tree_method</dt>
<dd>Method for tree building. Valid choices are: clearcut, clustalw, raxml, fasttree_v1, fasttree, muscle [default: fasttree]</dd>
<dt>-o, <cite>-</cite>-result_fp</dt>
<dd>Path to store result file [default: &lt;input_sequences_filename&gt;.tre]</dd>
<dt>-l, <cite>-</cite>-log_fp</dt>
<dd>Path to store log file [default: No log file created.]</dd>
<dt>-r, <cite>-</cite>-root_method</dt>
<dd>Method for choosing root of phylo tree  Valid choices are: midpoint, tree_method_default [default: tree_method_default]</dd>
</dl>
</div>
<p><strong>Output:</strong></p>
<p>The result of <a class="reference external" href="./make_phylogeny.html">make_phylogeny.py</a> consists of a newick formatted tree file (.tre) and optionally a log file. The tree file is formatted using the Newick format and this file can be viewed using most tree visualization tools, such as TopiaryTool, FigTree, etc.</p>
<p><strong>Examples:</strong></p>
<p>A simple example of <a class="reference external" href="./make_phylogeny.html">make_phylogeny.py</a> is shown by the following command, where we use the default tree building method (fasttree) and write the file to the current working directory without a log file:</p>
<div class="highlight-python"><pre>make_phylogeny.py -i repr_set_seqs_aligned_pfiltered.fasta -o rep_phylo.tre</pre>
</div>
<p>Alternatively, if the user would prefer using another tree building method (i.e. clearcut (Sheneman, Evans, &amp; Foster, 2006)), then they could use the following command:</p>
<div class="highlight-python"><pre>make_phylogeny.py -i repr_set_seqs_aligned_pfiltered.fasta -t clearcut</pre>
</div>
<p>Note: For whichever method used, the 3rd party program must be properly installed on the user&#8217;s computer.</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">

    


            <h4>Previous topic</h4>
            <p class="topless"><a href="parallel_align_seqs_pynast.html"
                                  title="previous chapter"><em>parallel_align_seqs_pynast.py</em> &#8211; Parallel sequence alignment using PyNAST</a></p>
            <h4>Next topic</h4>
            <p class="topless"><a href="assign_taxonomy.html"
                                  title="next chapter"><em>assign_taxonomy.py</em> &#8211; Assign taxonomy to each sequence</a></p>
            <h3>This Page</h3>
            <ul class="this-page-menu">
              <li><a href="../_sources/scripts/make_phylogeny.txt"
                     rel="nofollow">Show Source</a></li>
            </ul>
          <div id="searchbox" style="display: none">
            <h3>Quick search</h3>
              <form class="search" action="../search.html" method="get">
                <input type="text" name="q" size="18" />
                <input type="submit" value="Go" />
                <input type="hidden" name="check_keywords" value="yes" />
                <input type="hidden" name="area" value="default" />
              </form>
              <p class="searchtip" style="font-size: 90%">
              Enter search terms or a module, class or function name.
              </p>
          </div>
          <script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="assign_taxonomy.html" title="assign_taxonomy.py – Assign taxonomy to each sequence"
             >next</a> |</li>
        <li class="right" >
          <a href="parallel_align_seqs_pynast.html" title="parallel_align_seqs_pynast.py – Parallel sequence alignment using PyNAST"
             >previous</a> |</li>
        <li><a href="../index.html">Home</a> &raquo;</li>
          <li><a href="../documentation.html" >QIIME Documentation</a> &raquo;</li>
          <li><a href="../doc_script_index.html" >Scripts - Analyses and Parameters</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
      &copy; Copyright 2010, QIIME Team.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 0.6.3.
    </div>
  </body>
</html>