<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Introduction &mdash; Homepage</title>
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.9',
        COLLAPSE_MODINDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/favicon.ico"/>
    <link rel="top" title="Homepage" href="index.html" />
    <link rel="up" title="QIIME Documentation" href="documentation.html" />
    <link rel="next" title="Input Files" href="doc_input_files.html" />
    <link rel="prev" title="QIIME Documentation" href="documentation.html" /> 
  </head>
  <body>

<div style="background-color: white; text-align: left; padding: 0px">
<a href="index.html"><img src="_static/wordpressheader.png" border="0" alt="sampledoc"/></a>
</div>
<div class="news" style="background-color:#ccc;">
	<p style="font-size:12pt;padding: 5px;"><b>News and Announcements &raquo;</b>
<script type="text/javascript" src="http://app.feed.informer.com/digest3/U9OLMKHF9N.js"><noscript><a href="http://app.feed.informer.com/digest3/U9OLMKHF9N.html">Click for &quot;QIIME&quot;.</a> Powered by <a href="http://feed.informer.com/">RSS Feed Informer</a></noscript></script></p>
</div>

    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="doc_input_files.html" title="Input Files"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="documentation.html" title="QIIME Documentation"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Home</a> &raquo;</li>
          <li><a href="documentation.html" accesskey="U">QIIME Documentation</a> &raquo;</li> 
      </ul>
    </div>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <ul class="simple">
</ul>
<div class="section" id="introduction">
<h1>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h1>
<div class="section" id="who-should-use-this-software">
<h2>Who Should Use this Software?<a class="headerlink" href="#who-should-use-this-software" title="Permalink to this headline">¶</a></h2>
<p>This software is designed for scientists investigating the microbial diversity within and between samples using SSU (16S or 18S) rRNA gene sequence surveys where the gene sequences are generated by PCR and (mostly) sequenced with 454 pyrosequencing. In order to analyze the diversity of sequences within multiple samples with a single pyrosequencer run, barcodes are assigned to samples, and the amplicons are pooled for sequencing. This software recognizes the barcodes and sorts sequences back to their original samples, then applies a comprehensive set of analysis tools to assess the diversity within samples and compare samples to one another. The analysis also incorporates data provided by the user, to relate known properties of samples or sets of samples to their microbial constituents.</p>
<p>This software is designed for use by scientists familiar with a basic approach to the analysis of microbial diversity. Familiarity with the concepts of OTUs, phylogenetic trees, diversity estimates, and methods of community comparisons such as UniFrac and PCoA, are necessary for the interpretation of the results. The software is designed for wide use, is user-friendly, and does not require specialized skills. it is designed for students and professionals, or anyone wishing to analyze a microbial diversity dataset.</p>
</div>
<div class="section" id="getting-the-software">
<h2>Getting the Software<a class="headerlink" href="#getting-the-software" title="Permalink to this headline">¶</a></h2>
<p>The QIIME software can be found on Sourceforge at the following location: <a class="reference external" href="http://qiime.sourceforge.net/">http://qiime.sourceforge.net/</a></p>
</div>
<div class="section" id="microbial-community-analysis">
<h2>Microbial Community Analysis<a class="headerlink" href="#microbial-community-analysis" title="Permalink to this headline">¶</a></h2>
<p>This software is designed to generate lists of OTUs from very large datasets (mostly those generated by pyrosequencing), and to perform phylogenetic and nonphylogenetic OTU-based analyses. It performs sequence quality checks and chimera-checking, chooses OTUs, performs Sequence-, UniFrac- and OTU-based clustering of samples from which the sequences were obtained, assigns sequences to microbial lineages, displays heatmaps of common and rare OTUs in samples, performs rarefaction analysis for an assessment of depth of coverage, builds very large trees for tree-based community comparisons such as UniFrac, and applies a network-based analysis.</p>
</div>
<div class="section" id="sequencing-technologies">
<h2>Sequencing Technologies<a class="headerlink" href="#sequencing-technologies" title="Permalink to this headline">¶</a></h2>
<p>Software is in principle compatible with all platforms: 454, Illumina, and/or Sanger sequencing. Different technologies require different strategies for building the tree: for short reads or for heterogeneous reads that don&#8217;t overlap, it is more effective to insert the reads into a tree built using full-length sequences. This is because trees built with reads of less than 200 nucleotides are often highly inaccurate (though still often sufficient for tree-based community comparisons such as UniFrac), and trees generally cannot be built de novo with sequences that do not overlap at all. For combining heterogeneous datasets, the best approach is usually to use split_libraries.py on each dataset separately, combine the FASTA files, make an OTU table using BLAST (rather than cd-hit), and then use the combined OTU table and a combined sample mapping file for downstream analysis. Another important consideration is error rate: pyrosequencing reads that are not denoised (see (Quince et al., 2009)) can lead to very large numbers of artifactual OTUs, even at relatively high levels. However, this noise problem affects alpha diversity estimates (estimates of diversity within a sample) far more than beta diversity estimates (diversity across samples, e.g. clustering with UniFrac). See (Reeder &amp; Knight, 2009) for a commentary on this issue.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">

    
            <h3><a href="index.html">Table Of Contents</a></h3>
            <ul>
<li><a class="reference external" href="">Introduction</a><ul>
<li><a class="reference external" href="#who-should-use-this-software">Who Should Use this Software?</a></li>
<li><a class="reference external" href="#getting-the-software">Getting the Software</a></li>
<li><a class="reference external" href="#microbial-community-analysis">Microbial Community Analysis</a></li>
<li><a class="reference external" href="#sequencing-technologies">Sequencing Technologies</a></li>
</ul>
</li>
</ul>



            <h4>Previous topic</h4>
            <p class="topless"><a href="documentation.html"
                                  title="previous chapter">QIIME Documentation</a></p>
            <h4>Next topic</h4>
            <p class="topless"><a href="doc_input_files.html"
                                  title="next chapter">Input Files</a></p>
            <h3>This Page</h3>
            <ul class="this-page-menu">
              <li><a href="_sources/doc_introduction.txt"
                     rel="nofollow">Show Source</a></li>
            </ul>
          <div id="searchbox" style="display: none">
            <h3>Quick search</h3>
              <form class="search" action="search.html" method="get">
                <input type="text" name="q" size="18" />
                <input type="submit" value="Go" />
                <input type="hidden" name="check_keywords" value="yes" />
                <input type="hidden" name="area" value="default" />
              </form>
              <p class="searchtip" style="font-size: 90%">
              Enter search terms or a module, class or function name.
              </p>
          </div>
          <script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="doc_input_files.html" title="Input Files"
             >next</a> |</li>
        <li class="right" >
          <a href="documentation.html" title="QIIME Documentation"
             >previous</a> |</li>
        <li><a href="index.html">Home</a> &raquo;</li>
          <li><a href="documentation.html" >QIIME Documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
      &copy; Copyright 2010, QIIME Team.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 0.6.3.
    </div>
  </body>
</html>