<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Scripts - Analyses and Parameters &mdash; Homepage</title>
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.9',
        COLLAPSE_MODINDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/favicon.ico"/>
    <link rel="top" title="Homepage" href="index.html" />
    <link rel="up" title="QIIME Documentation" href="documentation.html" />
    <link rel="next" title="alpha_rarefaction.py – A workflow script for performing alpha rarefaction" href="scripts/alpha_rarefaction.html" />
    <link rel="prev" title="Input Files" href="doc_input_files.html" /> 
  </head>
  <body>

<div style="background-color: white; text-align: left; padding: 0px">
<a href="index.html"><img src="_static/wordpressheader.png" border="0" alt="sampledoc"/></a>
</div>
<div class="news" style="background-color:#ccc;">
	<p style="font-size:12pt;padding: 5px;"><b>News and Announcements &raquo;</b>
<script type="text/javascript" src="http://app.feed.informer.com/digest3/U9OLMKHF9N.js"><noscript><a href="http://app.feed.informer.com/digest3/U9OLMKHF9N.html">Click for &quot;QIIME&quot;.</a> Powered by <a href="http://feed.informer.com/">RSS Feed Informer</a></noscript></script></p>
</div>

    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="scripts/alpha_rarefaction.html" title="alpha_rarefaction.py – A workflow script for performing alpha rarefaction"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="doc_input_files.html" title="Input Files"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Home</a> &raquo;</li>
          <li><a href="documentation.html" accesskey="U">QIIME Documentation</a> &raquo;</li> 
      </ul>
    </div>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="scripts-analyses-and-parameters">
<h1>Scripts - Analyses and Parameters<a class="headerlink" href="#scripts-analyses-and-parameters" title="Permalink to this headline">¶</a></h1>
<div class="section" id="general-notes">
<h2>General Notes<a class="headerlink" href="#general-notes" title="Permalink to this headline">¶</a></h2>
<p>All QIIME analyses are performed using python (.py) scripts, which are located in the Qiime/scripts directory. To access QIIME python scripts, it may be useful to set an environment variable to the location of the innermost QIIME directory (the one containing <a class="reference external" href="./scripts/check_id_map.html">check_id_map.py</a>, for example):</p>
<div class="highlight-python"><pre>qdir=/path/to/QIIME/</pre>
</div>
<p>Further commands of the form <tt class="docutils literal"><span class="pre">python</span> <span class="pre">QIIME_script.py</span> <span class="pre">-o</span> <span class="pre">option</span></tt> can be invoked as <tt class="docutils literal"><span class="pre">python</span> <span class="pre">$qdir/QIIME_script.py</span> <span class="pre">-o</span> <span class="pre">option</span></tt>. For all path description throughout the Documentation, the &#8220;/path/to/&#8221; refers to the physical location of each program on your local computer. For instance, the &#8220;/path/to/QIIME/&#8221; on my computer refers to &#8220;/Users/Jesse/Qiime/&#8221; on Mac OS X version 10.5.</p>
<p>The user can obtain help about the arguments which can be passed to each script, as well as examples and usage notes, by typing the following in the bash shell:</p>
<div class="highlight-python"><pre>python $qdir/script_of_interest.py –h</pre>
</div>
</div>
<div class="section" id="workflow-scripts">
<h2>Workflow Scripts<a class="headerlink" href="#workflow-scripts" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/alpha_rarefaction.html"><em>alpha_rarefaction.py</em> &#8211; A workflow script for performing alpha rarefaction</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/beta_diversity_through_3d_plots.html"><em>beta_diversity_through_3d_plots.py</em> &#8211; A workflow script for computing beta diversity distance matrices and the corresponding 3D plots</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/jackknifed_upgma.html"><em>jackknifed_upgma.py</em> &#8211; A workflow script for performing jackknifed UPGMA clustering</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/pick_otus_through_otu_table.html"><em>pick_otus_through_otu_table.py</em> &#8211; A workflow script for picking OTUs through building OTU tables</a></li>
</ul>
</div>
<div class="section" id="user-generated-mapping-file">
<h2>User-Generated Mapping File<a class="headerlink" href="#user-generated-mapping-file" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/check_id_map.html"><em>check_id_map.py</em> &#8211; Checks user&#8217;s metadata mapping file for required data, valid format</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/merge_mapping_files.html"><em>merge_mapping_files.py</em> &#8211; Merge mapping files</a></li>
</ul>
</div>
<div class="section" id="quality-checking">
<h2>Quality Checking<a class="headerlink" href="#quality-checking" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/denoise.html"><em>denoise.py</em> &#8211; Denoise a flowgram file</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/identify_chimeric_seqs.html"><em>identify_chimeric_seqs.py</em> &#8211; Identify chimeric sequences in input FASTA file</a></li>
</ul>
</div>
<div class="section" id="demultiplexing">
<h2>Demultiplexing<a class="headerlink" href="#demultiplexing" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/split_libraries.html"><em>split_libraries.py</em> &#8211; Split libraries according to barcodes specified in mapping file</a></li>
</ul>
</div>
<div class="section" id="read-otu-file-otu-mapping">
<h2>Read OTU File (OTU Mapping)<a class="headerlink" href="#read-otu-file-otu-mapping" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/add_taxa.html"><em>add_taxa.py</em> &#8211; Add taxa to OTU table</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/filter_otus_by_sample.html"><em>filter_otus_by_sample.py</em> &#8211; Filter OTU mapping file and sequences by SampleIDs</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/merge_otu_maps.html"><em>merge_otu_maps.py</em> &#8211; Merge OTU mapping files</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_pick_otus_blast.html"><em>parallel_pick_otus_blast.py</em> &#8211; Parallel pick otus using BLAST</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/pick_otus.html"><em>pick_otus.py</em> &#8211; OTU picking</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/summarize_taxa.html"><em>summarize_taxa.py</em> &#8211; Summarize Taxa</a></li>
</ul>
</div>
<div class="section" id="process-sequences">
<h2>Process Sequences<a class="headerlink" href="#process-sequences" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/adjust_seq_orientation.html"><em>adjust_seq_orientation.py</em> &#8211; Get the reverse complement of all sequences</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/exclude_seqs_by_blast.html"><em>exclude_seqs_by_blast.py</em> &#8211; Exclude contaminated sequences using BLAST</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/extract_seqs_by_sample_id.html"><em>extract_seqs_by_sample_id.py</em> &#8211; Extract sequences based on the SampleID</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/pick_rep_set.html"><em>pick_rep_set.py</em> &#8211; Pick representative set of sequences</a></li>
</ul>
</div>
<div class="section" id="sequence-alignment">
<h2>Sequence Alignment<a class="headerlink" href="#sequence-alignment" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/align_seqs.html"><em>align_seqs.py</em> &#8211; Align sequences using a variety of alignment methods</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/filter_alignment.html"><em>filter_alignment.py</em> &#8211; Filter sequence alignment by removing highly variable regions</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_align_seqs_pynast.html"><em>parallel_align_seqs_pynast.py</em> &#8211; Parallel sequence alignment using PyNAST</a></li>
</ul>
</div>
<div class="section" id="tree-building">
<h2>Tree-Building<a class="headerlink" href="#tree-building" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/make_phylogeny.html"><em>make_phylogeny.py</em> &#8211; Make Phylogeny</a></li>
</ul>
</div>
<div class="section" id="taxonomy-assignment">
<h2>Taxonomy Assignment<a class="headerlink" href="#taxonomy-assignment" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/assign_taxonomy.html"><em>assign_taxonomy.py</em> &#8211; Assign taxonomy to each sequence</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_assign_taxonomy_blast.html"><em>parallel_assign_taxonomy_blast.py</em> &#8211; Parallel taxonomy assignment using BLAST</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_assign_taxonomy_rdp.html"><em>parallel_assign_taxonomy_rdp.py</em> &#8211; Parallel taxonomy assignment using RDP</a></li>
</ul>
</div>
<div class="section" id="otu-table-processing">
<h2>OTU Table Processing<a class="headerlink" href="#otu-table-processing" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/filter_otu_table.html"><em>filter_otu_table.py</em> &#8211; Filters OTU table by minimum OTU count and number of samples or by taxonomy</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/filter_by_metadata.html"><em>filter_by_metadata.py</em> &#8211; Filter OTU table by removal of specified metadata</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_otu_table.html"><em>make_otu_table.py</em> &#8211; Make OTU table</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/multiple_rarefactions.html"><em>multiple_rarefactions.py</em> &#8211; Perform rarefaction on multiple otu tables</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/otu_category_significance.html"><em>otu_category_significance.py</em> &#8211; OTU significance and co-occurence analysis</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_multiple_rarefactions.html"><em>parallel_multiple_rarefactions.py</em> &#8211; Parallel multiple file rarefaction</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/single_rarefaction.html"><em>single_rarefaction.py</em> &#8211; Perform rarefaction on a single otu table</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/summarize_otu_by_cat.html"><em>summarize_otu_by_cat.py</em> &#8211; Create a summarized OTU table for a specific metadata category</a></li>
</ul>
</div>
<div class="section" id="alpha-diversity">
<h2>Alpha-Diversity<a class="headerlink" href="#alpha-diversity" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/alpha_diversity.html"><em>alpha_diversity.py</em> &#8211; Calculate alpha diversity on each sample in an otu table, using a variety of alpha diversity metrics</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/alpha_diversity_metrics.html"><em>alpha_diversity_metrics</em> &#8211; List of available metrics</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/collate_alpha.html"><em>collate_alpha.py</em> &#8211; Collate alpha diversity results</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_rarefaction_averages.html"><em>make_rarefaction_averages.py</em> &#8211; Generate Rarefaction Averages</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_alpha_diversity.html"><em>parallel_alpha_diversity.py</em> &#8211; Parallel alpha diversity</a></li>
</ul>
</div>
<div class="section" id="beta-diversity">
<h2>Beta-Diversity<a class="headerlink" href="#beta-diversity" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/beta_diversity_metrics.html"><em>beta_diversity_metrics</em> &#8211; List of available metrics</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/beta_diversity.html"><em>beta_diversity.py</em> &#8211; Calculate beta diversity (pairwise sample dissimilarity) on one or many otu tables</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/dissimilarity_mtx_stats.html"><em>dissimilarity_mtx_stats.py</em> &#8211; Calculate mean, median and standard deviation from a set of distance matrices</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_beta_diversity.html"><em>parallel_beta_diversity.py</em> &#8211; Parallel beta diversity</a></li>
</ul>
</div>
<div class="section" id="principal-coordinates-analysis-pcoa">
<h2>Principal Coordinates Analysis (PCoA)<a class="headerlink" href="#principal-coordinates-analysis-pcoa" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/principal_coordinates.html"><em>principal_coordinates.py</em> &#8211; Principal Coordinates Analysis (PCoA)</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/transform_coordinate_matrices.html"><em>transform_coordinate_matrices.py</em> &#8211; Transform 2 coordinate matrices</a></li>
</ul>
</div>
<div class="section" id="visualization">
<h2>Visualization<a class="headerlink" href="#visualization" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/cytoscape_usage.html"><em>cytoscape_usage</em> &#8211; Loading Results with Cytoscape</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/compare_3d_plots.html"><em>compare_3d_plots.py</em> &#8211; Plot two PCoA files on the same 3D plot</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_2d_plots.html"><em>make_2d_plots.py</em> &#8211; Make 2D PCoA Plots</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_3d_plots.html"><em>make_3d_plots.py</em> &#8211; Make 3D PCoA plots</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_3d_plot_prefs_file.html"><em>make_3d_plot_prefs_file.py</em> &#8211; Generate preferences file for 3D plots using Metadata Fields</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_distance_histograms.html"><em>make_distance_histograms.py</em> &#8211; Make distance histograms</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_pie_charts.html"><em>make_pie_charts.py</em> &#8211; Make pie charts based on taxonomy assignment</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_otu_heatmap_html.html"><em>make_otu_heatmap_html.py</em> &#8211; Make heatmap of OTU table</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_otu_network.html"><em>make_otu_network.py</em> &#8211; Make an OTU network and calculate statistics</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_rarefaction_plots.html"><em>make_rarefaction_plots.py</em> &#8211; Generate Rarefaction Plots</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/preferences_file.html"><em>preferences_file</em> &#8211; Example of a prefs file</a></li>
</ul>
</div>
<div class="section" id="jackknifing">
<h2>Jackknifing<a class="headerlink" href="#jackknifing" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/make_bootstrapped_tree.html"><em>make_bootstrapped_tree.py</em> &#8211; Make bootstrapped tree</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/tree_compare.html"><em>tree_compare.py</em> &#8211; Compare jackknifed/bootstrapped trees</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/upgma_cluster.html"><em>upgma_cluster.py</em> &#8211; Build a UPGMA tree comparing samples</a></li>
</ul>
</div>
<div class="section" id="sra-submission">
<h2>SRA Submission<a class="headerlink" href="#sra-submission" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/make_fastq.html"><em>make_fastq.py</em> &#8211; Make fastq file for ERA submission from paired fasta and qual files</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_library_id_lists.html"><em>make_library_id_lists.py</em> &#8211; Make library id lists</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_per_library_sff.html"><em>make_per_library_sff.py</em> &#8211; Make per-library sff files from id lists</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_sra_submission.html"><em>make_sra_submission.py</em> &#8211; Makes SRA submission files (xml-format)</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/per_library_stats.html"><em>per_library_stats.py</em> &#8211; Calculate per library statistics</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/process_sff.html"><em>process_sff.py</em> &#8211; Convert sff to FASTA and QUAL files</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/sra_spreadsheet_to_map_files.html"><em>sra_spreadsheet_to_map_files.py</em> &#8211; Create mapping file from SRA submission spreadsheet</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/trim_sff_primers.html"><em>trim_sff_primers.py</em> &#8211; Trim sff primers</a></li>
</ul>
</div>
<div class="section" id="utilities">
<h2>Utilities<a class="headerlink" href="#utilities" title="Permalink to this headline">¶</a></h2>
<ul>
<li class="toctree-l1"><a class="reference external" href="scripts/blast_wrapper.html"><em>blast_wrapper.py</em> &#8211; Blast Interface</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/cluster_jobs.html"><em>cluster_jobs.py</em> &#8211; Cluster jobs</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/convert_unifrac_sample_mapping_to_otu_table.html"><em>convert_unifrac_sample_mapping_to_otu_table.py</em> &#8211; Convert a UniFrac sample mapping file to an OTU table</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/fix_arb_fasta.html"><em>fix_arb_fasta.py</em> &#8211; Reformat ARB FASTA files</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_qiime_py_file.html"><em>make_qiime_py_file.py</em> &#8211; Create python file</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/make_qiime_rst_file.html"><em>make_qiime_rst_file.py</em> &#8211; Make Sphinx RST file</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/parallel_blast.html"><em>parallel_blast.py</em> &#8211; Parallel BLAST</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/poller.html"><em>poller.py</em> &#8211; Poller for parallel QIIME scripts.</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/poller_example.html"><em>poller_example.py</em> &#8211; Create python file</a></li>
<li class="toctree-l1"><a class="reference external" href="scripts/print_qiime_config.html"><em>print_qiime_config.py</em> &#8211; Print out the qiime config settings.</a></li>
</ul>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">

    
            <h3><a href="index.html">Table Of Contents</a></h3>
            <ul>
<li><a class="reference external" href="">Scripts - Analyses and Parameters</a><ul>
<li><a class="reference external" href="#general-notes">General Notes</a></li>
<li><a class="reference external" href="#workflow-scripts">Workflow Scripts</a><ul>
</ul>
</li>
<li><a class="reference external" href="#user-generated-mapping-file">User-Generated Mapping File</a><ul>
</ul>
</li>
<li><a class="reference external" href="#quality-checking">Quality Checking</a><ul>
</ul>
</li>
<li><a class="reference external" href="#demultiplexing">Demultiplexing</a><ul>
</ul>
</li>
<li><a class="reference external" href="#read-otu-file-otu-mapping">Read OTU File (OTU Mapping)</a><ul>
</ul>
</li>
<li><a class="reference external" href="#process-sequences">Process Sequences</a><ul>
</ul>
</li>
<li><a class="reference external" href="#sequence-alignment">Sequence Alignment</a><ul>
</ul>
</li>
<li><a class="reference external" href="#tree-building">Tree-Building</a><ul>
</ul>
</li>
<li><a class="reference external" href="#taxonomy-assignment">Taxonomy Assignment</a><ul>
</ul>
</li>
<li><a class="reference external" href="#otu-table-processing">OTU Table Processing</a><ul>
</ul>
</li>
<li><a class="reference external" href="#alpha-diversity">Alpha-Diversity</a><ul>
</ul>
</li>
<li><a class="reference external" href="#beta-diversity">Beta-Diversity</a><ul>
</ul>
</li>
<li><a class="reference external" href="#principal-coordinates-analysis-pcoa">Principal Coordinates Analysis (PCoA)</a><ul>
</ul>
</li>
<li><a class="reference external" href="#visualization">Visualization</a><ul>
</ul>
</li>
<li><a class="reference external" href="#jackknifing">Jackknifing</a><ul>
</ul>
</li>
<li><a class="reference external" href="#sra-submission">SRA Submission</a><ul>
</ul>
</li>
<li><a class="reference external" href="#utilities">Utilities</a><ul>
</ul>
</li>
</ul>
</li>
</ul>



            <h4>Previous topic</h4>
            <p class="topless"><a href="doc_input_files.html"
                                  title="previous chapter">Input Files</a></p>
            <h4>Next topic</h4>
            <p class="topless"><a href="scripts/alpha_rarefaction.html"
                                  title="next chapter"><em>alpha_rarefaction.py</em> &#8211; A workflow script for performing alpha rarefaction</a></p>
            <h3>This Page</h3>
            <ul class="this-page-menu">
              <li><a href="_sources/doc_script_index.txt"
                     rel="nofollow">Show Source</a></li>
            </ul>
          <div id="searchbox" style="display: none">
            <h3>Quick search</h3>
              <form class="search" action="search.html" method="get">
                <input type="text" name="q" size="18" />
                <input type="submit" value="Go" />
                <input type="hidden" name="check_keywords" value="yes" />
                <input type="hidden" name="area" value="default" />
              </form>
              <p class="searchtip" style="font-size: 90%">
              Enter search terms or a module, class or function name.
              </p>
          </div>
          <script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="scripts/alpha_rarefaction.html" title="alpha_rarefaction.py – A workflow script for performing alpha rarefaction"
             >next</a> |</li>
        <li class="right" >
          <a href="doc_input_files.html" title="Input Files"
             >previous</a> |</li>
        <li><a href="index.html">Home</a> &raquo;</li>
          <li><a href="documentation.html" >QIIME Documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
      &copy; Copyright 2010, QIIME Team.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 0.6.3.
    </div>
  </body>
</html>